test/Atmos/EDMF/ekman_layer.jl

```diff
@@ -0,0 +1,271 @@
+    # Change boolean to control vertical discretization type
+    finite_volume = false
+
+    # Choice of compressibility and CFL
+    # compressibility = Compressible()
+    compressibility = Anelastic1D()
+    str_comp = compressibility == Compressible() ? "COMPRESS" : "ANELASTIC"
+
+    # Choice of SGS model
+    turbconv = NoTurbConv()
+    # N_updrafts = 1
+    # N_quad = 3
+    # turbconv = EDMF(FT, N_updrafts, N_quad, param_set)
+
+    C_smag_ = C_smag(param_set)
+    turbulence = ConstantKinematicViscosity(FT(0.1))
+    # turbulence = SmagorinskyLilly{FT}(C_smag_)
+
+    # Prescribe domain parameters
+    zmax = FT(400)
+    # Simulation time
+    t0 = FT(0)
+    timeend = FT(3600 * 2) # Change to 7h for low-level jet
+    CFLmax = compressibility == Compressible() ? FT(1) : FT(100)
+
+    config_type = SingleStackConfigType
+    ode_solver_type = ClimateMachine.ExplicitSolverType(
+        solver_method = LSRK144NiegemannDiehlBusch,
+    )
+
+    if finite_volume
+        N = (1, 0)
+        nelem_vert = 80
+        ref_state = HydrostaticState(
+            DryAdiabaticProfile{FT}(param_set),
+            ;
+            subtract_off = false,
+        )
+        output_prefix = string("EL_", str_comp, "_FVM")
+        fv_reconstruction = FVLinear()
+    else
+        N = 4
+        nelem_vert = 20
+        ref_state = HydrostaticState(DryAdiabaticProfile{FT}(param_set),)
+        output_prefix = string("EL_", str_comp, "_DG")
+        fv_reconstruction = nothing
+    end
+
+    surface_flux = cl_args["surface_flux"]
+    model = ekman_layer_model(
+        FT,
+        config_type,
+        zmax,
+        surface_flux;
+        turbulence = turbulence,
+        turbconv = turbconv,
+        ref_state = ref_state,
+        compressibility = compressibility,
+    )
+    # Assemble configuration
+    driver_config = ClimateMachine.SingleStackConfiguration(
+        output_prefix,
+        N,
+        nelem_vert,
+        zmax,
+        param_set,
+        model;
+        hmax = FT(40),
+        solver_type = ode_solver_type,
+        fv_reconstruction = fv_reconstruction,
+    )
+    solver_config = ClimateMachine.SolverConfiguration(
+        t0,
+        timeend,
+        driver_config,
+        init_on_cpu = true,
+        Courant_number = CFLmax,
+    )
+
+    # --- Zero-out horizontal variations:
+    vsp = vars_state(model, Prognostic(), FT)
+    horizontally_average!(
+        driver_config.grid,
+        solver_config.Q,
+        varsindex(vsp, :turbconv),
+    )
+    horizontally_average!(
+        driver_config.grid,
+        solver_config.Q,
+        varsindex(vsp, :energy, :ρe),
+    )
+    vsa = vars_state(model, Auxiliary(), FT)
+    horizontally_average!(
+        driver_config.grid,
+        solver_config.dg.state_auxiliary,
+        varsindex(vsa, :turbconv),
+    )
+    # ---
+
+    dgn_config = config_diagnostics(driver_config)
+
+    # boyd vandeven filter
+    cb_boyd = GenericCallbacks.EveryXSimulationSteps(1) do
```

Comment:
is this used?

Reply:
No, but I want to leave it there so we can easily plug it in the callback. This is also useful for issue #2116